elf/module.go

```diff
@@ -504,20 +503,29 @@ func (b *Module) closeSocketFilters() error {
 	return nil
 }
 
-func unpinMap(m *Map) error {
-	if m.m.def.pinning == 0 {
-		return nil
+func unpinMap(m *Map, pinPath string) error {
+	mapPath, err := getMapPath(&m.m.def, m.Name, pinPath)
+	if err != nil {
+		return err
 	}
-	namespace := C.GoString(&m.m.def.namespace[0])
-	mapPath := filepath.Join(BPFFSPath, namespace, BPFDirGlobals, m.Name)
 	return syscall.Unlink(mapPath)
 }
 
 func (b *Module) closeMaps(options map[string]CloseOptions) error {
 	for _, m := range b.maps {
 		doUnpin := options[fmt.Sprintf("maps/%s", m.Name)].Unpin
-		if m.m.def.pinning > 0 && doUnpin {
-			unpinMap(m)
+		if doUnpin {
+			mapDef := m.m.def
+			if mapDef.pinning == PIN_CUSTOM_NS {
+				pinPath := options[fmt.Sprintf("maps/%s", m.Name)].PinPath
```

Comment:
could you check if the entry in the map exists before dereferencing it with `.PinPath`?
```
closeOption, ok := options[fmt.Sprintf("maps/%s", m.Name)]
if !ok {
  return error...
}
pinPath := closeOption.PinPath
```